lib/dma.c

```diff
@@ -246,6 +246,12 @@ dma_controller_add_region(dma_controller_t *dma,
                        fd, offset, offset + size, region->fd);
                 goto err;
             }
+            if (region->prot != prot) {
+                vfu_log(dma->vfu_ctx, LOG_ERR, "bad prot=%#x "
+                        "for new DMA region %#lx-%#lx, existing prot=%#x\n",
+                        prot, offset, offset + size, region->prot);
```

Comment:
Region should end at `offset + size - 1`.

Reply:
Ack